src/osx/cocoa/dataview.mm

```diff
@@ -2552,6 +2545,72 @@ void UpdateWithRow(int row)
     [m_OutlineView editColumn:column row:[m_OutlineView rowForItem:[m_DataSource getDataViewItemFromBuffer:item]] withEvent:nil select:YES];
 }
 
+#if wxUSE_DRAG_AND_DROP
+
+bool wxCocoaDataViewControl::EnableDropTarget(const wxDataFormatArray& formats)
+{
+    [m_OutlineView unregisterDraggedTypes];
+
+    if (formats.GetCount() > 0)
+    {
+        wxCFMutableArrayRef<CFStringRef> typesarray;
+        for (size_t i = 0; i < formats.GetCount(); ++i)
+            // formats.Item(i).AddSupportedTypes(typesarray);
+        {
+            switch (formats.Item(i).GetType())
+            {
+                case wxDF_TEXT:
+                case wxDF_OEMTEXT:
+                case wxDF_UNICODETEXT:
+                    CFArrayAppendValue(typesarray, NSStringPboardType);
+                    break;
+
+                case wxDF_BITMAP:
+                    wxGCC_WARNING_SUPPRESS(deprecated-declarations)
+                    CFArrayAppendValue(typesarray, NSPICTPboardType);
+                    wxGCC_WARNING_RESTORE(deprecated-declarations)
+                    break;
+
+                case wxDF_FILENAME:
+                    CFArrayAppendValue(typesarray, NSFilenamesPboardType);
+                    break;
+
+                case wxDF_HTML:
+                    CFArrayAppendValue(typesarray, NSHTMLPboardType);
+                    break;
+
+                case wxDF_TIFF:
+                    CFArrayAppendValue(typesarray, NSTIFFPboardType);
+                    break;
+
+                case wxDF_METAFILE:
+                case wxDF_SYLK:
+                case wxDF_DIF:
+                case wxDF_DIB:
+                case wxDF_PALETTE:
+                case wxDF_PENDATA:
+                case wxDF_RIFF:
+                case wxDF_WAVE:
+                case wxDF_ENHMETAFILE:
+                case wxDF_LOCALE:
+                case wxDF_PRIVATE:
+                case wxDF_INVALID:
+                case wxDF_MAX:
+                    break;
+            }
+        }
+
+        // Add support for internal dataView items' DnD?
+        // CFArrayAppendValue(typesarray, DataViewPboardType);
```

Comment:
Should we enable `DataViewPboardType`? Maybe when enabling support for text types? `DataViewPboardType` is used in `writeItems` (line 849) when item's drag begin (BTW, I quite not understand the meaning of its use. Maybe Is it used for dragging items outside the app?

Reply:
`DataViewPboardType` is the private data type (=string constant) of the dataview implementation. It has the wxDataFormat ID in its first bytes, then the raw data

Reply:
> `DataViewPboardType` is the private data type (=string constant) of the dataview implementation. It has the wxDataFormat ID in its first bytes, then the raw data, only used within the app / wx apps themselves